### PR TITLE
Currency network test adapter

### PR DESCRIPTION
**Pull request by berndbohmeier**
WIP Use currency network adapter

Add a test adapter for the currency network to make the test easiert to
understand and easier to maintain.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -10,9 +10,122 @@
...
+                interest_rate_received,
+                is_frozen,
+                m_time,
+                balance,
+            ]
+        ):
+            if value is not None:
+                if value != account_tuple[index]:
+                    return False
+            return True
```
Comment (cducrest) [inline tests/conftest.py]: It's hard to see for me but I don't think this is properly indented

File: tests/conftest.py [hunk 2]
```diff
@@ -10,9 +10,122 @@
...
+            self.contract.functions.updateTrustline(
+                creditor_address,
+                creditline_received,
+                creditline_given,
+                interest_rate_received,
+                interest_rate_given,
+                is_frozen,
+            ).transact({"from": debtor_address})
+
+    def transfer(self, value: int, *, path, max_fee=0, extra_data=EXTRA_DATA):
```
Comment (cducrest) [inline tests/conftest.py]: I guess a default of max_uint64 would be better for max_fee.

Comment (berndbohmeier) [reply tests/conftest.py]: makes sense

File: tests/conftest.py (hunk 1 above)
Comment (berndbohmeier) [reply tests/conftest.py]: yes, you are right. That happens when you dont use the code you write immediately. I'll make sure that this is used, and fix it.

File: tests/conftest.py (hunk 2 above)
Comment (berndbohmeier) [reply tests/conftest.py]: Makes sense

**Pull request merged**